### ISSUE TITLE
Datagrid extension: Customize columns modal can become unusable for zoom users

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-cloud-cognitive#2465]
## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

> Describe in detail the issue you're having.

When zooming to 150-200%, the scrollable area of the customize columns can become very small or inaccessible entirely. This makes it difficult 

> Is this issue related to a specific component?

Modal in datagrid extension for customizing columns

> What did you expect to happen? What happened instead? What would you like to
> see changed?

More of the modal area should remain scrollable/viewable at small screen sizes. Possible solutions may include a smaller header area, unsticking sections, or a small screen friendlier approach for search to allow more real estate for the content.

> What browser are you working in?

Chrome, Safari

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?

1.35.2

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

## Steps to reproduce the issue

1. Navigate to Storybook
2. Go to customize columns story (`Datagrid` > `Extensions` > `ColumnCustomization`)
3. Open customize column modal from datagrid actions.
4. Increase zoom levels to 150-200% (depending on screen sizes).

> Please create a reduced test case in CodeSandbox
>
> https://codesandbox.io/s/github/carbon-design-system/ibm-cloud-cognitive/tree/main/examples/cloud-cognitive/codesandbox

## Additional information

Initially reported by Mike Gower.

![image.png](https://images.zenhubusercontent.com/6109a215337f0517b2f0e02f/87b5c760-9965-47da-b111-eb0a340ac14a)

- Screenshots or code
- Notes

**Comment by elycheea** [comment]
Looks to be related to modals themselves — we should check if Carbon already has an issue open for this.

**Comment by elycheea** [comment]
Closing for now since the modal will be moved to `Tearsheet` in #2290, which leverages more vertical real estate than the default modals.
